### PR TITLE
build,test: various fixes to align with upstream

### PR DESCRIPTION
**Pull request by kfarnung**
While preparing the v8.x branch I noticed a few oddities in the node.gyp file.  This change attempts to bring the file more closely into alignment with upstream.

I also removed an orphaned test status entry.

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

build, test

### CODE REVIEW

**Comment by kfarnung** [comment]
CI: https://ci.nodejs.org/job/chakracore-test-pull-request/180/

**Review by digitalinfinity — APPROVED** [review]
Seems reasonable

File: node.gyp [hunk 1]
```diff
@@ -870,7 +870,7 @@
             'deps/v8/include'
           ],
           'conditions' : [
-            ['node_use_v8_platform=="true" and node_engine=="v8"', {
+            ['node_use_v8_platform=="true"', {
```
Comment (jackhorton) [inline node.gyp]: I remember we had an issue where we were using v8_platform incorrectly -- I think @digitalinfinity looked into it? Wasn't this the gtest issue?

**Review by jackhorton — APPROVED** [review]

File: node.gyp (hunk 1 above)
Comment (digitalinfinity) [reply node.gyp]: Oh yeah, nice catch! @kfarnung https://github.com/nodejs/node-chakracore/pull/431 is the relevant PR

Comment (kfarnung) [reply node.gyp]: @digitalinfinity this one is already embedded inside a `node_engine=="v8"` condition, so this should be unnecessary.  I think the bug you were seeing was caused by the unnecessary duplicate entry:

https://github.com/nodejs/node-chakracore/pull/446/files#diff-259e627be3fafed9e9c10a7fcb26879eL998

**Pull request closed without merging**